modules/rank-eval/src/test/java/org/elasticsearch/index/rankeval/QuerySpecTests.java

```diff
@@ -63,7 +70,58 @@ public static void afterClass() throws Exception {
         searchRequestParsers = null;
     }
 
-    // TODO add some sort of roundtrip testing like we have now for queries?
+    public static RatedRequest createTestItem(List<String> indices, List<String> types) {
+        String specId = randomAsciiOfLength(50);
+
+        SearchSourceBuilder testRequest = new SearchSourceBuilder();
+        testRequest.size(23);
+        testRequest.query(new MatchAllQueryBuilder());
+
+        List<RatedDocument> ratedDocs = new ArrayList<>();
+        int size = randomIntBetween(0, 2);
+        for (int i = 0; i < size; i++) {
+            ratedDocs.add(RatedDocumentTests.createTestItem());
+        }
+        
+        return new RatedRequest(specId, testRequest, indices, types, ratedDocs);
+    }
+
+    public void testXContentRoundtrip() throws IOException {
+        List<String> indices = new ArrayList<>();
+        int size = randomIntBetween(0, 20);
+        for (int i = 0; i < size; i++) {
+            indices.add(randomAsciiOfLengthBetween(0, 50));
+        }
+
+        List<String> types = new ArrayList<>();
+        size = randomIntBetween(0, 20);
+        for (int i = 0; i < size; i++) {
+            types.add(randomAsciiOfLengthBetween(0, 50));
+        }
+
+        RatedRequest testItem = createTestItem(indices, types);
+
+        XContentBuilder builder = XContentFactory.contentBuilder(randomFrom(XContentType.values()));
+        if (randomBoolean()) {
+            builder.prettyPrint();
+        }
+        testItem.toXContent(builder, ToXContent.EMPTY_PARAMS);
+        XContentBuilder shuffled = shuffleXContent(builder);
+        XContentParser itemParser = XContentHelper.createParser(shuffled.bytes());
+        itemParser.nextToken(); // TODO this could be the reason why the metric roundtrip tests failed
```

Comment:
Is this TODO still relevant? I ran the tests and they passed, or am I missing sth.?